File: apps/api/internal/provider/huggingface.go
package provider

import "github.com/manpreet/chatbot/apps/api/internal/config"

func NewHuggingFaceClient(cfg config.ChatConfig) ChatCompleter {
	return newCompatibleClient(
		"huggingface",
		"huggingface",
		cfg.HuggingFaceBaseURL,
		cfg.HuggingFaceAPIKey,
		cfg.Model,
	)
}

File: apps/pyapi/pyapi/providers/huggingface.py
from __future__ import annotations

from pyapi.config import ChatConfig

from .compatible import OpenAICompatibleProvider


class HuggingFaceProvider(OpenAICompatibleProvider):
    def __init__(self, config: ChatConfig):
        super().__init__(
            provider="huggingface",
            base_url=config.huggingface_base_url,
            api_key=config.huggingface_api_key,
            model=config.model,
            http_referer=config.http_referer,
            app_title=config.app_title,
        )
